var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -778,3 +802,106 @@ def check(self):
         with spack.util.environment.set_env(CP2K_DATA_DIR=data_dir, PWD=self.build_directory):
             with working_dir(self.build_directory):
                 make("test", *self.build_targets)
+
+
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    """Use the new cmake build system to build cp2k. It is the default when
+    building the master branch of cp2k."""
+
+    def cmake_args(self):
+        spec = self.spec
+
+        if "+cuda" in spec and len(spec.variants["cuda_arch"].value) > 1:
+            raise InstallError("CP2K supports only one cuda_arch at a time")
+
+        if "+rocm" in spec and len(spec.variants["amdgpu_target"].value) > 1:
+            raise InstallError("CP2K supports only one amdgpu_arch at a time")
+
+        args = []
+
+        if spec.satisfies("+cuda"):
+            cuda_arch = self.spec.variants["cuda_arch"].value[0]
+
+            gpu_map = {"35": "K40", "37": "K80", "60": "P100", "70": "V100", "80": "A100"}
+
+            gpuver = gpu_map[cuda_arch]
+
+            args += ["-DCP2K_WITH_GPU=%s" % gpuver, "-DCP2K_USE_ACCEL=cuda"]
+
+        if spec.satisfies("+rocm"):
+            amd_arch = self.spec.variants["amdgpu_target"].value[0]
+
+            gpuver = {"gfx906": "Mi50", "gfx908": "Mi100", "gfx90a": "Mi250x"}[amd_arch]
+
+            args += ["-DCP2K_WITH_GPU={0}".format(gpuver), "-DCP2K_USE_ACCEL=hip"]
+
+        args += [
+            self.define_from_variant("CP2K_USE_ELPA", "elpa"),
+            self.define_from_variant("CP2K_USE_LIBINT", "libint"),
+            self.define_from_variant("CP2K_USE_SIRIUS", "sirius"),
+            self.define_from_variant("CP2K_USE_SPLA", "spla"),
+            self.define_from_variant("CP2K_USE_COSMA", "cosma"),
+            self.define_from_variant("CP2K_USE_LIBXC", "libxc"),
+        ]
+
+        if spec.satisfies("+pytorch"):
+            args += ["-DCP2K_USE_LIBTORCH=ON"]
+
+        args += [
+            self.define_from_variant("CP2K_USE_METIS", "metis"),
+            self.define_from_variant("CP2K_USE_PLUMED", "plumed"),
+            self.define_from_variant("CP2K_USE_SPGLIB", "spglib"),
+            self.define_from_variant("CP2K_USE_VORI", "libvori"),
+            self.define_from_variant("CP2K_USE_SUPERLU", "superlu-dist"),
+            self.define_from_variant("CP2K_USE_SPLA", "spla"),
+            self.define_from_variant("CP2K_USE_QUIP", "quip"),
+        ]
+
+        # we force the use elpa openmp threading support. might need to be revisited though
+        if (spec.satisfies("+openmp") and spec.satisfies("+elpa")) or spec.satisfies(
+            "^elpa+openmp"
+        ):
+            args += ["-DCP2K_ENABLE_ELPA_OPENMP_SUPPORT=ON"]
+
+        if "spla" in spec and (spec.satisfies("+cuda") or spec.satisfies("+rocm")):
+            args += ["-DCP2K_USE_SPLA_GEMM_OFFLOADING=ON"]
+
+        if "fftw" in spec:
+            args += ["-DCP2K_USE_FFTW=ON"]
+
+        with when("smm=libxsmm"):
+            args += ["-DCP2K_USE_LIBXSMM=ON"]
```

Comment:
I have to check whether the following works:
```
self.define_from_variant("CP2K_USE_LIBXSMM", "smm=libxsmm"),
```

but in any case we might want to set it to OFF explicitly just to anticipate the CP2K CMake to switch the default at some point.
Also, in DBCSR CMake we have modeled that flag the same way as in in Spack: `-DUSE_SMM=...` to anticipate new/other libsmm providers.